### PR TITLE
istio: Update to release 1.1.3

### PR DESCRIPTION
**Pull request by jrajahalme**
Update the CI and the getting started guide to Istio 1.1.3.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

```release-note
Istio GSG is updated to Istio release 1.1.3.
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7789)

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-me-please

**Comment by ianvernon** [comment]
toServices flakes: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/11740/testReport/junit/Suite-k8s-1/14/K8sPolicyTest_Basic_Test_Validate_to_entities_policies_Validate_toEntities_Cluster/

Unrelated to this PR. Running again to get things green :-)

**Comment by ianvernon** [comment]
test-me-please

File: Documentation/gettingstarted/istio.rst [hunk 1]
```diff
@@ -140,464 +140,78 @@ Check the progress of the deployment (every service should have an
...
-    details-v1            1         1         1            1           22m
-    productpage-v2        1         1         1            1           4m
-    ratings-v1            1         1         1            1           19m
-    reviews-v1            1         1         1            1           22m
-    reviews-v2            1         1         1            1           19m
+   $ export GATEWAY_URL=$INGRESS_HOST:$INGRESS_PORT
 
-Check that the product REST API is still accessible, and that Cilium
-now denies at Layer-7 any access to the reviews and ratings REST API:
+- Confirn that the Bookinfo application is reachable from outside of the cluster:
```
Comment (ianvernon) [inline Documentation/gettingstarted/istio.rst]: Confirm

File: Documentation/gettingstarted/istio.rst [hunk 2]
```diff
@@ -140,464 +140,78 @@ Check the progress of the deployment (every service should have an
...
-``HTTP 403 Forbidden`` HTTP responses.
-
-Every login and logout on the product page will result in a line in
-this service's log. Note that you need to log in/out using the ``sign
-in``/``sign out`` element on the bookinfo web page. When you do, you
-can observe these kind of audit logs:
-
-::
-
-    $ export POD_LOGGER_V1=`kubectl get pods -l app=authaudit-logger,version=v1 -o jsonpath='{.items[0].metadata.name}'`
```
Comment (ianvernon) [inline Documentation/gettingstarted/istio.rst]: What happened to the authaudit logger?

**Review by ianvernon — CHANGES_REQUESTED** [review]

**Comment by jrajahalme** [comment]
Known CI flake on transparent encryption and VXLAN encapsulation (#7615).

**Comment by jrajahalme** [comment]
test-me-please

File: Documentation/gettingstarted/istio.rst (hunk 1 above)
Comment (jrajahalme) [reply Documentation/gettingstarted/istio.rst]: Thanks!

File: Documentation/gettingstarted/istio.rst (hunk 2 above)
Comment (jrajahalme) [reply Documentation/gettingstarted/istio.rst]: Did not get that far yet & wanted to see how bad we want it in the GSG...

**Comment by jrajahalme** [comment]
test-me-please

File: Documentation/gettingstarted/istio.rst (hunk 2 above)
Comment (jrajahalme) [reply Documentation/gettingstarted/istio.rst]: @ianvernon I rebased the whole existing GSG to Istio 1.1.3, please take a look.

**Comment by jrajahalme** [comment]
test-docs-please

File: Documentation/gettingstarted/istio.rst [hunk 3]
```diff
@@ -145,10 +145,13 @@ Step 3: Deploy the Bookinfo Application V1
 
 Now that we have Cilium and Istio deployed, we can deploy version
 ``v1`` of the services of the `Istio Bookinfo sample application
-<https://istio.io/docs/guides/bookinfo.html>`_.
+<https://istio.io/docs/examples/bookinfo.html>`_.
 
-The BookInfo application is broken into four separate microservices:
+While the upstream `Istio Bookinfo
+Applicatio example for Kubernetes
```
Comment (ianvernon) [inline Documentation/gettingstarted/istio.rst]: Applicatio --> Application

File: Documentation/gettingstarted/istio.rst [hunk 4]
```diff
@@ -145,10 +145,13 @@ Step 3: Deploy the Bookinfo Application V1
 
 Now that we have Cilium and Istio deployed, we can deploy version
 ``v1`` of the services of the `Istio Bookinfo sample application
-<https://istio.io/docs/guides/bookinfo.html>`_.
+<https://istio.io/docs/examples/bookinfo.html>`_.
 
-The BookInfo application is broken into four separate microservices:
+While the upstream `Istio Bookinfo
```
Comment (ianvernon) [inline Documentation/gettingstarted/istio.rst]: This line returns too early, fix the formatting.

File: Documentation/gettingstarted/istio.rst [hunk 5]
```diff
@@ -19,8 +19,8 @@ environment running on your machine.
 
    If running on minikube, you may need to up the memory and CPUs
    available to the minikube VM from the defaults and/or the
-   instructions provided here for the other GSGs. 6GB and 3 CPUs
-   should be enough for Istio (``--memory=6144 --cpus=3``).
+   instructions provided here for the other GSGs. 6GB and 4 CPUs
```
Comment (ianvernon) [inline Documentation/gettingstarted/istio.rst]: Why do we need 4 CPUs now vs. 3?

File: Documentation/gettingstarted/istio.rst [hunk 6]
```diff
@@ -205,16 +198,30 @@ Check the progress of the deployment (every service should have an
...
+    reviews-v1       1         1         1            1           2m
+
+Once all the pods are available, verify that the application works by
+making a query from reviews pod to the productpage:
+
+::
+
+   $ kubectl exec -it $(kubectl get pod -l app=reviews -o jsonpath='{.items[0].metadata.name}') -c istio-proxy -- curl productpage:9080/productpage | grep "<title>.*</title>"
+
+Create an Istio ingress gateway for the productpage service:
```
Comment (ianvernon) [inline Documentation/gettingstarted/istio.rst:215]: Why is the ordering different now for creating the Istio ingress gateway from before?

**Comment by jrajahalme** [comment]
test-docs-please

**Comment by jrajahalme** [comment]
test-me-please

File: test/k8sT/manifests/bookinfo-v1-istio.yaml [hunk 7]
```diff
@@ -172,7 +172,7 @@ spec:
         - "9080"
         - -d
         - "15020"
-        image: docker.io/istio/proxy_init:1.1.2
+        image: docker.io/istio/proxy_init:1.1.3
```
Comment (ianvernon) [inline test/k8sT/manifests/bookinfo-v1-istio.yaml:175]: As part of this PR, please do the following. 

* Open a PR against `packer-ci-build` and add the new images added in any `test/k8s/manifests/*.yaml` file that was changed as part of this PR. The file to change is `provision/pull-images.sh`. 
* Once that is done, build a new version of the box via logging into Jenkins, going to the [Vagrant-Master-Boxes-Packer-Build job](https://jenkins.cilium.io/job/Vagrant-Master-Boxes-Packer-Build/), and clicking `Build With Parameters` to build a new version of the box.
* Update the version of the VM used in `Vagrantfile` and `test/Vagrantfile`.

This ensures that the VMs have the images cached for use in the CI.

**Review by ianvernon — CHANGES_REQUESTED** [review]

File: Documentation/gettingstarted/istio.rst (hunk 3 above)
Comment (jrajahalme) [reply Documentation/gettingstarted/istio.rst]: Fixed, thanks!

File: Documentation/gettingstarted/istio.rst (hunk 4 above)
Comment (jrajahalme) [reply Documentation/gettingstarted/istio.rst]: Done.

File: Documentation/gettingstarted/istio.rst (hunk 5 above)
Comment (jrajahalme) [reply Documentation/gettingstarted/istio.rst]: It was needed when I ran the upstream yaml that loaded both v1 and v2 at the same time. We are not using the upstream yaml any more, will check again on 3 cpus (and 5GB memory) to see if tighter resources still work.

File: Documentation/gettingstarted/istio.rst (hunk 6 above)
Comment (jrajahalme) [reply Documentation/gettingstarted/istio.rst:215]: This is the ordering used in the upstream example, and it makes sense to test the intra-service connectivity first, before adding in the external connectivity. Specifically, while doing this rebase I noticed that the prior method of getting the external URL did not work any more, and I needed to figure out where the problem was. Internal connectivity was fine, but had that not been fine,  would have wondered about the ingress gateway URL for nothing.

File: Documentation/gettingstarted/istio.rst (hunk 5 above)
Comment (jrajahalme) [reply Documentation/gettingstarted/istio.rst]: With 3 CPUs, the step where ratings-v1 and reviews-v2 are added fails to schedule both of them, with this warning in the `kubectl pod describe`:

`  Warning  FailedScheduling  3m (x2 over 3m)  default-scheduler  0/1 nodes are available: 1 Insufficient cpu.`

I'll try again with 4 CPUs and 5GB of memory.

Comment (jrajahalme) [reply Documentation/gettingstarted/istio.rst]: 4 CPUs works with 5 GB, updated accordingly.

File: test/k8sT/manifests/bookinfo-v1-istio.yaml (hunk 7 above)
Comment (jrajahalme) [reply test/k8sT/manifests/bookinfo-v1-istio.yaml:175]: Packer CI PR created (pr/jrajahalme/istio-update-1.1.3). Apparently I do not have build rights on this repo, though.

**Comment by jrajahalme** [comment]
Streamlined the GSG for better usability by removing duplicate command lines, and splitting some multi-command-line sequences into individual blocks, as this makes copying the command line from the browser much easier (copy-first-line button).

**Comment by jrajahalme** [comment]
vagrant VM box yet to be updated, pending packer-ci-build build (for which I do not have the access rights).

File: test/k8sT/manifests/bookinfo-v1-istio.yaml (hunk 7 above)
Comment (ianvernon) [reply test/k8sT/manifests/bookinfo-v1-istio.yaml:175]: I merged your PR, and built the image :D 

https://jenkins.cilium.io/job/Vagrant-Master-Boxes-Packer-Build/147/

You can update the Vagrantfiles to use version 147.

**Comment by jrajahalme** [comment]
Rebase to master, Vagrant boxes updated.

File: test/k8sT/manifests/bookinfo-v1-istio.yaml (hunk 7 above)
Comment (jrajahalme) [reply test/k8sT/manifests/bookinfo-v1-istio.yaml:175]: Done!

**Comment by jrajahalme** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Comment by jrajahalme** [comment]
test-me-please

**Comment by ianvernon** [comment]
Build failed: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/11766/execution/node/83/log/

hit #7797 .

**Comment by ianvernon** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**